scanpipe/pipes/d2d.py

```diff
@@ -548,13 +638,51 @@ def match_purldb(project, extensions, matcher_func, logger=None):
             increment_percent=10,
             start_time=start_time,
         )
-        matched_package = matcher_func(project, to_resource)
-        if matched_package:
-            matched_count += 1
 
     logger(f"{matched_count:,d} resource(s) matched in PurlDB")
 
 
+def match_purldb_directories(project, logger=None):
+    """Match against PurlDB selecting codebase directories."""
+    to_directories = (
+        project.codebaseresources.directories().to_codebase().order_by("path")
```

Comment:
Could you add a comment about the need for the ordering in the context of the directories matching?